lib/Request.js

```diff
@@ -30,8 +30,7 @@ function Request(method, url, params, opts, tries, callback) {
 
         if (opts.ignore_http_errors !== true &&
             res                     !== undefined &&
-            res.statusCode          >=  400   &&
-            res.statusCode          <=  500) {
+            isErrorStatusCode(res.statusCode, opts.error_status_codes)) {
```

Comment:
Seems like a lot of string parsing to do on each request, perhaps we can pre-process to an array or better yet hash table lookup.